From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Mon, 10 Apr 2023 14:25 +0100

When -i used, the fips_run_rng_test() should not be called in
update_kernel_random().

Upstream PR #189, already merged in master.

---

diff -aur a/rngd.c b/rngd.c
--- a/rngd.c
+++ b/rngd.c
@@ -728,10 +728,11 @@
 	unsigned char *buf, fips_ctx_t *fipsctx_in)
 {
 	unsigned char *p;
-	int fips;
+	int fips = 0;
 	int rc;
 
-	fips = fips_run_rng_test(fipsctx_in, buf);
+	if (!arguments->ignorefail)
+		fips = fips_run_rng_test(fipsctx_in, buf);
 	if (fips && !arguments->ignorefail)
 		return 1;
 
